pkgs/applications/editors/neovim/utils.nix

```diff
@@ -215,35 +217,33 @@ let
         (lib.removePrefix "tree-sitter-")
         (lib.replaceStrings [ "-" ] [ "_" ])
       ];
+
+      toNvimTreesitterGrammar = callPackage ({ gnused }:
+        makeSetupHook {
+          name = "to-nvim-treesitter-grammar";
+          propagatedBuildInputs = [ gnused ];
+          substitutions = {
+            sed = "${gnused}/bin/sed";
+          };
+        } ./to-nvim-treesitter-grammar.sh) {};
     in
 
-    toVimPlugin (runCommand "treesitter-grammar-${name}"
-      {
-        meta = {
-          platforms = lib.platforms.all;
-        } // grammar.meta;
-      }
-      ''
-        mkdir -p "$out/parser"
-        ln -s "${grammar}/parser" "$out/parser/${name}.so"
-
-        mkdir -p "$out/queries/${name}"
-        if [ -d "${grammar}/queries/${name}" ]; then
-          echo "moving queries from neovim queries dir"
-          for file in "${grammar}/queries/${name}"*; do
-            ln -s "$file" "$out/queries/${name}/$(basename "$file")"
-          done
-        else
-          if [ -d "${grammar}/queries" ]; then
-            echo "moving queries from standard queries dir"
-            for file in "${grammar}/queries/"*; do
-              ln -s "$file" "$out/queries/${name}/$(basename "$file")"
-            done
-          else
-            echo "missing queries for ${name}"
-          fi
-        fi
-      '');
+    (toVimPlugin (stdenv.mkDerivation {
+      name = "treesitter-grammar-${name}";
+
+      origGrammar = grammar;
+      grammarName = name;
+      installQueries = false;
```

Comment:
Should this be in `passthru`? I don't really know what `passthru` is for to be honest.

Reply:
`passthru` is for settings, that are not available during build, and thus don't trigger a rebuild. In Vim plugins it's used for dependencies, because you don't have plugin dependencies during the build (it only downloads a GitHub repo and reorganizes some files)

https://nixos.org/manual/nixpkgs/unstable/#chap-passthru

Reply:
Thanks for the pointer. Here the build needs all of these variables, so passthru can't be used.